### DISCUSSION TITLE
Model statistics

### DISCUSSION THREAD

**Discussion by BorisKourt** [google/model-viewer#3489] [Q&A] [answered]
I wasn't able to find if there is any internal implementation for this. 

Is there a way to get some basic statistics for the model, like draw calls, polygon count, and number of materials?

I have been using the frontend version of: https://github.com/donmccurdy/glTF-Transform but it is overkill for this, and doesn't have some statistics like draw calls. 

Edit: answer provided in this comment: https://github.com/google/model-viewer/discussions/3489#discussioncomment-2813844

**Comment by samaneh-kazemi** [comment, marked as answer]
Hi @BorisKourt, thanks for reaching out. You can use our editor. https://modelviewer.dev/editor/

Once you select or load a model, click on the info button under **Validation Report**. We use [KhronosGroup/glTF-Validator](https://github.com/KhronosGroup/glTF-Validator/) to give such statistics. 


<img width="1326" alt="Screen Shot 2022-05-24 at 9 42 08 AM" src="https://user-images.githubusercontent.com/3915613/170089002-e87ca982-462a-4bab-9112-1c6dc9519670.png">
<img width="1296" alt="Screen Shot 2022-05-24 at 9 43 21 AM" src="https://user-images.githubusercontent.com/3915613/170088992-3925fe56-dfd5-4a08-ab8e-b7b432844992.png">

**Reply by BorisKourt** [reply]
Thanks! I was hoping to be able to do it programmatically. Since I have model-viewer embedded in my application. I will try to dig through your implementation of  [KhronosGroup/glTF-Validator](https://github.com/KhronosGroup/glTF-Validator/) wasn't able to get it to work myself. 

Is the editor in this repo as well?

**Reply by elalish** [reply]
yes, it's the `space-opera` package. I believe this file should help you: https://github.com/google/model-viewer/blob/master/packages/space-opera/src/components/model_viewer_snippet/components/validation_utils.ts

**Reply by BorisKourt** [reply]
Thanks you both!
